### PR TITLE
feat: expose style for interpolation fit functions

### LINKED ISSUE #1489: Expose fit function style
**Describe the solution you'd like**
The user may desire to style differently the fitted values depending on their use cases.
We currently hardcoded some values, like the dashed style and the opacity value of the area.
We should expose, through the theme, a way to add a custom style for those rendered lines/areas/points.

**Describe alternatives you've considered**
We probably don't want to include a new fit function to linearly connect all points with a straight line and no area opacity.
Exposing the style is way come useful and the user can also reuse that style in different fitting functions.

### PR DESCRIPTION
**Pull request by markov00**
## Summary

This PR exposes the style options for the currently available interpolation fit functions.
In particular:
- the `area` path of an interpolated area can be customized as:
```ts
type AreaFitStyle = 
  Visible & // hide or show the area path
  Opacity & { // change the opacity used to fill the area
    fill: Color | typeof ColorVariant.Series;  // a specific color or the assigned data series color
    texture?: TexturedStyles; // an optional texture
  };
```
Screenshot 2021-11-25 at 17 51 14

rendered with:
```tsx
<AreaSeries
  areaSeriesStyle={{
    fit: {
      line: { visible: false },
      area: {
        fill: 'rgba(33, 33, 33, 0.3)',
        texture: { opacity: 0.2, stroke: 'gray', rotation: -45, shape: TextureShape.Line },
      },
    },
  }}
```

- the `line` path can be customized with:
```ts
type LineFitStyle =
  Visible & // hide or show the line path
  Opacity & // change the opacity used to render the line stroke
  StrokeDashArray & { // an optional dash stroke array
    stroke: Color | typeof ColorVariant.Series; // a specific color or the assigned data series color
  };
```
Screenshot 2021-11-25 at 17 48 14

rendered with:
```tsx
<LineSeries
    lineSeriesStyle={{
      fit: {
        line: {
          dash: [2, 10, 10, 5],
          stroke: 'rgba(0, 0, 0, 0.2)',
        },
      },
  }}
```

## Details

The style is exposed through the `LineSeriesStyle` and the `AreaSeriesStyle` so it can be configured globally and/or configured per series.

I take the chance to refactor a bit the code, removing some optional arguments and reversing small logic to improve code readability.

cc @elastic/eui-design
Please double-check the `theme` property.  We basically just expose the currently used style without breaking changes.
The visualization editor will take care of handling and applying a different style when requested by the user (generally they are just requesting to show a straight line and a filled area without any differences between the interpolated line/area and the original line/area created with the dataset.

## Issues

fix  #1489 

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] The `:theme` label has been added and the `@elastic/eui-design` team has been pinged when there are `Theme` API changes
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated
- [x] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
- [x] Visual changes have been tested with all available themes including `dark`, `light`, `eui-dark` & `eui-light`

### CODE REVIEW

File: packages/charts/src/utils/geometry.ts [hunk 1]
```diff
@@ -116,13 +116,12 @@ export interface LineGeometry {
     y: number;
   };
   seriesIdentifier: XYChartSeriesIdentifier;
-  seriesLineStyle: LineStyle;
-  seriesPointStyle: PointStyle;
+  style: LineSeriesStyle;
```
Comment (nickofthyme) [inline packages/charts/src/utils/geometry.ts:119]: Nice 👍

File: packages/charts/src/utils/themes/theme.ts [hunk 2]
```diff
@@ -532,15 +532,36 @@ export interface BubbleSeriesStyle {
...
+  Opacity & {
+    color: Color | typeof ColorVariant.Series;
+    texture?: TexturedStyles;
+  };
+
+/** @public */
+export type LineFitStyle = Visible &
+  Opacity &
+  StrokeDashArray & {
+    color: Color | typeof ColorVariant.Series;
```
Comment (nickofthyme) [inline packages/charts/src/utils/themes/theme.ts:562]: Do you think this should be `stroke` to match the naming of other styles?

**Review by nickofthyme — APPROVED** [review]
LGTM, I just made two comments on the api

File: packages/charts/src/utils/themes/theme.ts (hunk 2 above)
Comment (markov00) [reply packages/charts/src/utils/themes/theme.ts]: done https://github.com/elastic/elastic-charts/pull/1505/commits/57cc053dc00ce82f7253e3e478842c26d27878d1

File: packages/charts/src/utils/themes/theme.ts [hunk 3]
```diff
@@ -532,15 +532,36 @@ export interface BubbleSeriesStyle {
...
+  fit: {
+    line: LineFitStyle;
+    area: AreaFitStyle;
+  };
 }
 
+/** @public */
+export type AreaFitStyle = Visible &
+  Opacity & {
+    color: Color | typeof ColorVariant.Series;
```
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/utils/themes/theme.ts]: done https://github.com/elastic/elastic-charts/pull/1505/commits/57cc053dc00ce82f7253e3e478842c26d27878d1

**Review by rshen91 — APPROVED** [review]
These changes LGTM - tested locally

**Review by nickofthyme — APPROVED** [review]

**Pull request merged**